packages/sd-jwt/src/action-handler.ts

```diff
@@ -70,28 +69,43 @@ export class SDJwtPlugin implements IAgentPlugin {
 
   /**
    * Get the key to sign the SD-JWT
-   * @param issuer - did url like did:exmaple.com#key-1
+   * @param identifier - identifier like a did and other forms of identifiers
    * @param context - agent instance
    * @returns the key to sign the SD-JWT
    */
-  private async getSignKey(issuer: string, context: IRequiredContext) {
-    debug(`Getting signing key for issuer ${issuer}`)
-    const identifier = await context.agent.didManagerGet({
-      did: issuer.split('#')[0],
-    })
-    const doc = await mapIdentifierKeysToDocWithJwkSupport({ identifier, vmRelationship: 'assertionMethod' }, context)
-    if (!doc || doc.length === 0) {
-      throw new Error('No key found for signing')
-    }
-    const key = doc.find((key) => key.meta.verificationMethod.id === issuer)
-    if (!key) {
-      throw new Error(`No key found with the given id: ${issuer}`)
+  async getSignKey (
+      identifier: string,
+      context: IRequiredContext
+      //todo make the type of key known
+  ): Promise<{ alg: string; key: any }> {
+    if (identifier.startsWith('did:')) {
+      const didDocument = await context.agent.resolveDid({ didUrl: identifier });
+      const key = didDocument.verificationMethod?.find(vm => vm.publicKeyJwk);
+      if (!key) {
+        throw new Error(`No verification key found for DID: ${identifier}`);
+      }
+      const alg = key.publicKeyJwk.alg;
+      return { alg, key };
+    } else if (identifier.includes('x5c')) {
+      const x5cKey = await context.agent.getCertificateChain({ identifier });
```

Comment:
A cetificate chain is far from a key. It is an array of certificates in DER format, which in turn contains a public key.

You will only want to get the public key matched against an IKey of the agent for the certificate that is at the end/or beginning (I always forget the order of the x5c). So you would need to extract the key from the DER (or PEM). I am working on X509 related functionalities. But I rather have a big fixme and error thrown, then this code which isn't remotely what it should do

Reply:
refactored the x5c handling part. I wrote a couple of tests in our crypto-extensions to make sure this will work